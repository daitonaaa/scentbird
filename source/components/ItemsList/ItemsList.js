import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import classNames from 'classnames';

import './ItemsList.scss';

import Item from 'components/Item';


export class ItemsList extends Component {

  static propTypes = {
    list: PropTypes.array.isRequired,
    count: PropTypes.number.isRequired,
    openedItem: PropTypes.number.isRequired,
  };

  render() {
    const {
      list,
      count,
      openedItem,
    } = this.props;

    if (list instanceof Array && list.length) return (
      <div className="items-list">
        <div
          className={classNames('items-list__count', {
            full: !!count,
          })}
        >
          Выделено дочерних элементов: <span>{count}</span>
        </div>
        {list.map(item =>
          <Item
            {...item}
            key={item.id}
            open={item.id === openedItem}
          />
        )}
      </div>
    );

    return (
      <div className="items-list">
        <div className="items-list__no-items">
          Элементы не найдены
        </div>
      </div>
    );
  }
}


const mapStateToProps = state => ({
  ...state.items
});

export default connect(
  mapStateToProps
)(ItemsList);
