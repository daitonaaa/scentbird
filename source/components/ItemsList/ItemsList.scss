@import '~vars.scss';

.items-list {
  padding: 15px 15px 15px 0;

  &__no-items {
    padding: 20px;
    font-family: $mainFontBold;
    font-size: 18px;
    color: $accent;
    text-align: center;
  }

  &__count {
    margin-bottom: 15px;
    font-size: 13px;
    line-height: 40px;
    color: #d4d4d4;
    text-align: center;
    border: solid 1px #f3f3f3;
    border-radius: 5px;
    transition: all ease .2s;

    span {
      font-family: $mainFontBold;
      font-size: 15px;
    }

    &.full {
      background: #e9ffef;
      color: #467153;
      border: solid 1px #89af95;
    }
  }
}
