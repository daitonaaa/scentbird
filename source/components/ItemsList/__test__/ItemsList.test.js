import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ItemsList } from 'components/ItemsList/ItemsList';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


describe('> > > COMPONENT - ItemsList', () => {
  it('#create snapshot', () => {
    const output = shallow(
      <ItemsList
        list={[]}
        count={0}
        openedItem={0}
      />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('#test render items && render count prop', () => {
    const list = [
      { id: 1 }, { id: 2 }, { id: 3 }
    ];
    const output = shallow(
      <ItemsList
        count={21}
        openedItem={0}
        list={list}
      />
    );

    expect(output.find('.items-list__count')).toBeDefined();
    expect(output.find('.items-list__count span').text()).toBe('21');
  });
});
