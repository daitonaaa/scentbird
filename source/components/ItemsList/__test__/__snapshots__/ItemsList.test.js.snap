// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`> > > COMPONENT - ItemsList #create snapshot 1`] = `
<div
  className="items-list"
>
  <div
    className="items-list__no-items"
  >
    Элементы не найдены
  </div>
</div>
`;
